watchtower/server/server_test.go

```diff
@@ -0,0 +1,634 @@
+// +build dev
```

Comment:
Are the build tags necessary here? As is, a simple `go test -v` in this directory won't execute these tests. We can't always assume that the user is using our set of makefile tools. Seems this is just to have access to the mocked variants of the major interfaces?

Reply:
the linter (and other go tools) complain because the mock classes aren't accessible w/o the dev tag, it is nice that putting them behind a flag keeps mock structs from being rendered on godoc.org

can't help those that do not follow the install docs :P a sizable portion of the tests won't pass/compile without the `dev` tag anyway, seems they're already out of luck